frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1890,26 +1891,66 @@
             }
         };
 
-        vm.showMdDialog = function(ev, submissionId) {
-            for (var i = 0; i < vm.submissionResult.count; i++) {
-                if (vm.submissionResult.results[i].id === submissionId) {
-                    vm.submissionMetaData = vm.submissionResult.results[i];
-                    break;
+        vm.isOptionChecked = function (option, attribute) {
+        if (
+            attribute.values.findIndex((el) => {
+            return el === option
+            }) !== -1
+        ) {
+            return true
+        }
+        return false
+        }
+
+        vm.showMdDialog = function (ev, submissionId) {
+        for (var i = 0; i < vm.submissionResult.count; i++) {
+            if (vm.submissionResult.results[i].id === submissionId) {
+            vm.submissionMetaData = vm.submissionResult.results[i]
+            break
+            }
+        }
+        vm.method_name = vm.submissionMetaData.method_name
+        vm.method_description = vm.submissionMetaData.method_description
+        vm.project_url = vm.submissionMetaData.project_url
+        vm.publication_url = vm.submissionMetaData.publication_url
+        vm.submissionId = submissionId
+        if (vm.submissionMetaAttributes != null) {
+            if (vm.submissionMetaAttributes.length > 0) {
+            vm.submissionMetaAttributeIndex = vm.submissionMetaAttributes.findIndex(
+                (el) => {
+                return el.phaseId === vm.phaseId
+                },
+            )
+            vm.submissionMetaAttributes[
+                vm.submissionMetaAttributeIndex
+            ].attributes.forEach((attribute, index) => {
+                if (vm.submissionMetaData.submission_metadata === null) {
+                attribute.value = null
+                attribute.values = []
+                } else if (
```

Comment:
Please fix the indentation the code you have written. It is inconsistent right now. You can refer to other functions of this file to understand how we are indenting the frontend codebase.

Reply:
@Ram81 which extension should I use for automatic code formatting? I am facing trouble in setting up the formatting config according to the project. Every extension I use, makes change to the whole code

Reply:
@Ram81 I am avoiding formatting it manually and am trying to setup the formatting config by giving in time once, so that it helps me in my further contributions to the project, but am unable to find any extension. The docs say that html/js/css pretifier must be used but I could not find any extension with the exact same name

Reply:
I don't use extensions for code formatting. I just use VS code for my development work. Not sure which extension would be a good one to use.

Reply:
oh, I am too using VS Code as well.. could you please share with me your extension/configs from your settings so that our code can become consistent in terms of formatting

Reply:
>could you please share with me your extension/configs from your settings so that our code can become consistent in terms of formatting

I don't use any JS extensions, the only settings I have are to convert tabs to spaces.

Reply:
oh

Reply:
oh, i'll manually fix the code formatting then, thank you so much